### PR TITLE
Refac

### PR DESCRIPTION
**Pull request by Vighnesh-V**
Made changes to the linter to produce friendlier and ~aesthetic~ error messages! Take a look :)
Screen Shot 2020-04-10 at 12 35 41 AM
Screen Shot 2020-04-10 at 12 43 12 AM

### CODE REVIEW

**Comment by janeyx99** [comment]
Two nitpicky things:
* To make the fix a little clearer, maybe have "rewriting using && or ||"
* To be consistent, change "Instead of writing a one armed pattern match, consider using a let statement" to "using a let statement instead of an one-armed pattern match" (If you disagree with this change, still change "a one armed" to "an one-armed". I'm not sure what "one armed" should look like, but the dash will hopefully make it clearer.

File: README.md [hunk 1]
```diff
@@ -1,15 +1,20 @@
 # Camelot
-A fully-modular, ppx-based OCaml style linter.
+An OCaml Linter / Style Checker
 
 ## Dependencies 
 Install the following with `opam install` after creating a switch:
-- ocamlfind
+- ocamlfind (for debugging / dev)
+- ocamlbuild
 - ppx_tools
```
[the remark this answers is not in the document]
Comment (Vighnesh-V) [reply README.md:8]: It's for debugging! You can print ast fragments with `ocamlfind ppx_tools/dumpast - e "your ocaml expresson here"` / a  file argument.

File: checkers/simplebexp.ml [hunk 2]
```diff
@@ -7,35 +7,36 @@ open Report
 
 (* ------------------ Checks rule: if cond then true else false ------------------------- *)
 module IfReturnLit : CHECK = struct
-  let check ({location;code} : Report.lctxt) : warn option =
+
+  let fix = "replacing the if statement with just the condition"
+    
+  let check ({location;code; src} : Report.lctxt) : hint option =
```
[the remark this answers is not in the document]
Comment (Vighnesh-V) [reply checkers/simplebexp.ml:13]: `simplebexp.ml` and `simplepat.ml` don't actually work post refactor - I was going to use them as a way/starting point to get devfam/anyone interested into working with the linter!

File: checkers/simplebexp.ml [hunk 3]
```diff
@@ -7,35 +7,36 @@ open Report
...
+
 end
 
 (* ------------------ Checks rule: if cond then false else true ------------------------- *)
 module IfReturnLitInv : CHECK = struct
-  let check ({location;code} : lctxt) : warn option =
+
+  let fix = "replacing the if statement with (not condition)"
+    
+  let check ({location;code; src} : lctxt) : hint option =
```
[the remark this answers is not in the document]
Comment (Vighnesh-V) [reply checkers/simplebexp.ml:33]: see above

File: checkers/simpleeq.ml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+open Astutils
+
+module EqList : CHECK = struct
+  open Check
+  let fix = "using a pattern match to check whether a list has a certain value"
+
+  let check st {location; source; pattern} = 
+    begin match pattern with
+      | EqApply (lhs, rhs) ->
+        if is_list_literal lhs || is_list_literal rhs then
```
[the remark this answers is not in the document]
Comment (Vighnesh-V) [reply checkers/simpleeq.ml:12]: done!

File: checkers/simplepat.ml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+open Parsetree
+open Longident
+open Report
+open Check    
+
+module OneArmedMatch : CHECK = struct
+
+  let fix = "Instead of writing a one armed pattern match, consider using a let statement"
+    
+  let check ({location;code;src} : lctxt) : hint option =
```
[the remark this answers is not in the document]
Comment (Vighnesh-V) [reply checkers/simplepat.ml:11]: see comments above for simple pat

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (KeenWill) [reply README.md]: cool sounds good

**Pull request merged**